### PR TITLE
Update to 1.19, based on 1.18-fixup.

### PR DESCRIPTION
**Pull request by TheCurle**
Simple as it gets.

### CODE REVIEW

File: src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java [hunk 1]
```diff
@@ -127,7 +128,7 @@ public ExplosionMetaPackage explode(Level worldIn, BlockPos source, int stage, E
 
                 if (doBlocksDrop) {
                     ObjectArrayList<Pair<ItemStack, BlockPos>> objectarraylist = new ObjectArrayList<>();
-                    Collections.shuffle(meta.affectedBlocks, worldIn.random);
+                    Collections.shuffle(meta.affectedBlocks, new Random());
```
Comment (Unbekannt1998) [inline src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java]: why use new Random here and not use the one from the world?
Same for VanillaExplosionType

Comment (TheCurle) [reply src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java]: The world does not have a Random, it has a RandomSource which is Mojang's custom and wholly incompatible system.

Comment (dpeter99) [reply src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java]: Could a wrapper be made to make them compatible? If not we should at least base the new Random's seed on the world seed. (Global random from the world see and use those as the seed for these)
I feel like being deterministic, based on the world seed would be useful.

Comment (TheCurle) [reply src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java]: They are not compatible, and it is not possible to get the world seed.

RandomSource is a completely new system, it cannot be wrapped around Java's native Random, and it's seed is not visible even with reflection.

File: src/data/java/dev/socketmods/socketnukes/datagen/utils/advancment/AdvancementProvider.java [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import net.minecraft.advancements.Advancement;
+import net.minecraft.data.CachedOutput;
 import net.minecraft.data.DataGenerator;
 import net.minecraft.data.DataProvider;
 import net.minecraft.data.HashCache;
```
Comment (AterAnimAvis) [inline src/data/java/dev/socketmods/socketnukes/datagen/utils/advancment/AdvancementProvider.java]: No need to import `HashCache`, its no longer used, neither is the `GSON` field in the class

File: src/data/java/dev/socketmods/socketnukes/datagen/utils/loottable/BaseLootTableProvider.java [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import dev.socketmods.socketnukes.SocketNukes;
+import net.minecraft.data.CachedOutput;
 import net.minecraft.data.DataGenerator;
 import net.minecraft.data.DataProvider;
 import net.minecraft.data.HashCache;
```
Comment (AterAnimAvis) [inline src/data/java/dev/socketmods/socketnukes/datagen/utils/loottable/BaseLootTableProvider.java]: No need to import `HashCache`, its no longer used, neither is the `GSON` field in the class

File: src/main/java/dev/socketmods/socketnukes/block/explosive/TNTExplosive.java [hunk 4]
```diff
@@ -226,21 +227,20 @@ public InteractionResult use(BlockState state, Level worldIn, BlockPos pos, Play
...
-    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level p_153212_, BlockState p_153213_, BlockEntityType<T> p_153214_) {
-        return null;
+    public BlockEntity newBlockEntity(BlockPos blockPos, BlockState state) {
+        return new ExplosiveTileEntity(blockPos, state);
     }
 
     @org.jetbrains.annotations.Nullable
     @Override
-    public <T extends BlockEntity> GameEventListener getListener(Level p_153210_, T p_153211_) {
+    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level p_153212_, BlockState p_153213_, BlockEntityType<T> p_153214_) {
```
Comment (AterAnimAvis) [inline src/main/java/dev/socketmods/socketnukes/block/explosive/TNTExplosive.java:242]: No longer need to import `net.minecraft.world.level.gameevent.GameEventListener`

File: src/main/java/dev/socketmods/socketnukes/networking/packet/ExtendedExplosionPacket.java [hunk 5]
```diff
@@ -89,8 +89,8 @@ public void toBytes(FriendlyByteBuf buf) {
             buf.writeBlockPos(pos);
         }
 
-        buf.writeResourceLocation(SNRegistry.getName(this.explosionParticle));
-        buf.writeResourceLocation(SNRegistry.getName(this.explosionSound));
+        buf.writeResourceLocation(ForgeRegistries.PARTICLE_TYPES.getKey(this.explosionParticle));
```
Comment (AterAnimAvis) [inline src/main/java/dev/socketmods/socketnukes/networking/packet/ExtendedExplosionPacket.java:91]: Unused import, no longer using SNRegistry in this class.

File: src/main/java/dev/socketmods/socketnukes/registry/SNRegistry.java [hunk 6]
```diff
@@ -45,15 +45,15 @@ public class SNRegistry {
...
     /***********************************************
      *            SocketNukes Registries           *
      **********************************************/
 
-    public static Supplier<IForgeRegistry<ExtendedExplosionType>> EXPLOSION_TYPE_REGISTRY = EXPLOSIONS.makeRegistry("explosion_types", () ->
-            new RegistryBuilder<ExtendedExplosionType>().setMaxID(Integer.MAX_VALUE - 1).onAdd((owner, stage, id, obj, old) ->
-                SocketNukes.LOGGER.info("ExplosionType Added: " + getName(obj).toString() + " ")
+    public static Supplier<IForgeRegistry<ExtendedExplosionType>> EXPLOSION_TYPE_REGISTRY = EXPLOSIONS.makeRegistry(() ->
+            new RegistryBuilder<ExtendedExplosionType>().setMaxID(Integer.MAX_VALUE - 1).onAdd((owner, stage, id, key, obj, old) ->
+                SocketNukes.LOGGER.info("ExplosionType Added: " + key.location() + " ")
```
Comment (AterAnimAvis) [inline src/main/java/dev/socketmods/socketnukes/registry/SNRegistry.java:56]: Should this be reduced to debug level?

File: src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java (hunk 1 above)
Comment (AterAnimAvis) [reply src/main/java/dev/socketmods/socketnukes/explosion/types/CubicExplosionType.java]: Shouldn't we follow Vanilla here and migrate to `net.minecraft.Util#shuffle`?

**Review by AterAnimAvis — DISMISSED** [review]
Not introduced in this PR but I noticed that Explosions via the `Exploderisor 9000` and igniting a Popfilter are all aligned to the blockpos which produces noticable "jumps" in position, we may need to either expand our api slightly or look over our uses of it to instead center the explosions on the clicked block, as the current behaviour looks unnatural. All of which can wait for another PR/later.

The change to `ForgeRegistries.<REGISTRY>.getKey(SNRegistry.EXPLODER_ITEM.get())` from `SNRegistry.getName` whilst necessary does drop the `Objects.requireNonNull` which is a slight pain but nothing major (IDE warnings yeah!), Looking over the usages I'm tempted to introduce a helper class containing `SNName.for<Registry>`.

`BolbEntity#getEntityTexture` can go the way of the yeet

`pack.mcmeta` needs updating to match the changes in the mdk's <https://github.com/MinecraftForge/MinecraftForge/blob/1.19.x/mdk/src/main/resources/pack.mcmeta>

Overall 👍 apart from some nitpicks

**Pull request merged**